pkg/cosign/cosign.go

```diff
@@ -149,6 +150,11 @@ func VerifySignature(opts Options) (digest string, err error) {
 		return "", errors.Wrap(err, "issuer mismatch")
 	}
 
+	subject, _ := extractSubject(opts.ImageRef, payload, log)
+	if subject != "" && wildcard.Match(opts.Subject, subject) {
```

Comment:
this would skip the check for an empty subject....do we need to check `subject != ""`?

Reply:
Same comment for the issuer check, seems like we should remove ` issuer != ""` so the issuer has to match.

Reply:
removed check for empty string instead added `err` check to ensure `subject` and `issuer` values are present in SimpleContainerImage.Optional